### DISCUSSION TITLE
react decorator plugin issue

### DISCUSSION THREAD

**Discussion by peternish** [community/community#63403] [Codespaces]
### Select Topic Area

Bug

### Body

Hello. I am trying to use Tailwind CSS in my react app and it requires decorator plugin. 
I appended the following code in jsconfig.json file but it doesn't work. 
"allowSyntheticDefaultImports": true,
"experimentalDecorators": true
Any guru?

**Comment by DanTch** [comment]
Hello! I’m sorry to hear that you’re having trouble using the Tailwind CSS decorator plugin in your React app. From the information you’ve provided, it seems like you’ve added the "allowSyntheticDefaultImports": true and "experimentalDecorators": true options to your jsconfig.json file, but it’s not working as expected.

It’s important to note that the jsconfig.json file is used to configure the behavior of the JavaScript language service and does not affect the runtime behavior of your code. The "experimentalDecorators" option is used to enable support for decorators in your editor, but it does not enable support for decorators in your code.

To use decorators in your React app, you’ll need to configure Babel to transpile them. You can do this by installing the @babel/plugin-proposal-decorators package and adding it to your Babel configuration. Here’s an example of how you can do this:

Install the @babel/plugin-proposal-decorators package by running npm install --save-dev @babel/plugin-proposal-decorators.
In your Babel configuration file (usually .babelrc or babel.config.js), add the following:
{
  "plugins": [
    ["@babel/plugin-proposal-decorators", { "legacy": true }]
  ]
}
Copy
I hope this helps! Let me know if you have any further questions or if there’s anything else I can help with. 😊

**Comment by EnriquezAlberto** [comment]
To use Tailwind CSS with a React app, you don't typically need to enable TypeScript decorators. Instead, you need to set up Tailwind CSS properly in your project. Here's a basic outline of steps to get Tailwind CSS working in a React app:

1_ Install Dependencies: Make sure you have the required dependencies installed in your project.

`npm install tailwindcss postcss autoprefixer`

2_ Configure Tailwind CSS: Generate a configuration file for Tailwind CSS.

`npx tailwindcss init`

3_ Edit Tailwind Configuration: Customize the tailwind.config.js file to suit your project's needs.

4_ Import Tailwind CSS: In your main CSS file (e.g., src/index.css or src/index.scss), import Tailwind CSS styles.

```
@import 'tailwindcss/base';
@import 'tailwindcss/components';
@import 'tailwindcss/utilities';
```

5_ Use Tailwind CSS Classes: Start using Tailwind CSS classes in your React components.

6_ Add PurgeCSS (Optional, for Production): To remove unused CSS in production, you can set up PurgeCSS. Make sure to configure it properly in your build process.

7_ Run Your App: Start or build your React app as usual.

**Remember that Tailwind CSS doesn't require TypeScript decorators. It's primarily a utility-first CSS framework that allows you to quickly style your components using pre-defined classes.**

If you're encountering specific issues with your setup, it would be helpful to provide more details or error messages you're encountering. This will allow me to provide more targeted assistance, I'm here to help you :)

**Comment by peternish** [comment]
Thanks very much!
